src/components/Playground.svelte

```diff
@@ -0,0 +1,201 @@
+    editor.setValue(lzString.decompressFromEncodedURIComponent(compressedCode));
+  }
+
+  onMount(async () => {
+    worker = new CompilerWorker();
+    worker.onerror = (err) => console.error(err);
+
+    const urlParams = new URLSearchParams(window.location.search);
+    const code = lzString.decompressFromEncodedURIComponent(urlParams.get("code")!)
+      || 'module Playground\n\nprint("Hello, world!")\n';
+    editor = await createGrainEditor("editor", code);
+
+    const prefersDark = window.matchMedia("(prefers-color-scheme: dark)").matches;
+    themeChangeUnsub = store.$selectedTheme.subscribe(selectedTheme => {
+      monaco.editor.setTheme(selectedTheme === "light" || (!selectedTheme && !prefersDark) ? "github-light" : "github-dark-modified");
+    });
+
+    worker.addEventListener("message", ({ data }) => {
+      if (data?.initialized) {
+        compile();
+      }
+    });
+
+    editor.onDidChangeModelContent(() => {
+      const code = editor.getValue();
+      const compressed = lzString.compressToEncodedURIComponent(code);
+      const searchParams = new URLSearchParams({
+        code: compressed
+      });
+      window.history.replaceState(null, "", "?" + searchParams.toString());
+    });
+
+    function readjustForWidth() {
+      editor.layout();
+      if (window.innerWidth < 640) {
+        editor.updateOptions({ lineNumbers: "off" })
+      } else {
+        editor.updateOptions({ lineNumbers: "on" })
+      }
+    }
+
+    window.addEventListener(
+      "resize",
+      debounce(() => {
+        readjustForWidth();
+      }, 100)
+    );
+
+    readjustForWidth();
+  });
+
+  onDestroy(() => {
+    themeChangeUnsub();
+  });
+
+</script>
+
+<div class="overflow-x-hidden">
+  <div class="flex text-lg justify-between items-center px-4 lg:px-16 py-4 relative md:static">
+    <ButtonLink
+      onClick={compile}
+      disabled={loadingState}
+      className="w-full justify-center md:w-fit dark:disabled:bg-gray-variant-50 disabled:bg-[#9B9B9B] mr-4"
+    >
+      Run
+    </ButtonLink>
+    <div class="flex gap-6">
+      <div class="md:relative">
+        <button on:click={toggleExamples} class="peer cursor-pointer hover:text-color-accent md:px-6 py-2">
+          <span class="hidden md:inline-block">Examples</span>
+          <ListBulletsIcon class="text-color-dim md:hidden" />
+        </button>
+        <div class={`absolute z-10 left-0 md:left-1/2 md:-translate-x-1/2 ${examplesToggled ? "grid" : "hidden"} hover:grid peer-hover:grid grid-cols-2 px-2 py-2 rounded border border-gray-20 dark:border-purple-70 bg-white dark:bg-purple-90 w-screen md:w-96`}>
+          {#each codeExamples as example}
+            <button
+              on:click={switchCodeExample(lzString.compressToEncodedURIComponent(example.code))}
+              class="cursor-pointer hover:bg-gray-10 dark:hover:bg-purple-80 rounded px-4 py-2 text-left"
+            >
+              {example.name}
```

Comment:
I noticed when you try to run the `Practical` example the playground obviously errors because there are no files available, we might want to add a note about that in the playground or we can solve the problem in another way such as creating the file before reading it or preloading it into the wasi fs. I feel like we should avoid having any "program errors" if possible on our playground examples though.